packages/expo-file-system/ios/FileSystemModule.swift

```diff
@@ -134,18 +134,23 @@ public final class FileSystemModule: Module {
       try ensureFileDirectoryExists(localUrl)
       try ensurePathPermission(appContext, path: localUrl.path, flag: .write)
 
-      let session = options.sessionType == .background ? backgroundSession : foregroundSession
-      let request = createUrlRequest(url: sourceUrl, headers: options.headers)
-      let downloadTask = session.downloadTask(with: request)
-      let taskDelegate = EXSessionDownloadTaskDelegate(
-        resolve: promise.resolver,
-        reject: promise.legacyRejecter,
-        localUrl: localUrl,
-        shouldCalculateMd5: options.md5
-      )
-
-      sessionTaskDispatcher.register(taskDelegate, for: downloadTask)
-      downloadTask.resume()
+      if sourceUrl.isFileURL {
+        try ensurePathPermission(appContext, path: sourceUrl.path, flag: .read)
+        EXFileSystemLocalFileHandler.copy(from: sourceUrl, to: localUrl, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+      } else {
+        let session = options.sessionType == .background ? backgroundSession : foregroundSession
+        let request = createUrlRequest(url: sourceUrl, headers: options.headers)
+        let downloadTask = session.downloadTask(with: request)
+        let taskDelegate = EXSessionDownloadTaskDelegate(
+          resolve: promise.resolver,
+          reject: promise.legacyRejecter,
+          localUrl: localUrl,
+          shouldCalculateMd5: options.md5
+        )
+
+        sessionTaskDispatcher.register(taskDelegate, for: downloadTask)
+        downloadTask.resume()
+      }
```

Comment:
Let's avoid using `if/else` when there could be one `if` with a `return`. Then the rest of the code doesn't need an indent

```suggestion
      if sourceUrl.isFileURL {
        try ensurePathPermission(appContext, path: sourceUrl.path, flag: .read)
        EXFileSystemLocalFileHandler.copy(from: sourceUrl, to: localUrl, resolver: promise.resolver, rejecter: promise.legacyRejecter)
        return
      }
      let session = options.sessionType == .background ? backgroundSession : foregroundSession
      let request = createUrlRequest(url: sourceUrl, headers: options.headers)
      let downloadTask = session.downloadTask(with: request)
      let taskDelegate = EXSessionDownloadTaskDelegate(
        resolve: promise.resolver,
        reject: promise.legacyRejecter,
        localUrl: localUrl,
        shouldCalculateMd5: options.md5
      )

      sessionTaskDispatcher.register(taskDelegate, for: downloadTask)
      downloadTask.resume()
```